### ISSUE TITLE
raft: nil pointer dereference in raft.stepLeader

### ISSUE THREAD

**Issue by tschottdorf** [cockroachdb/cockroach#2950]
The following test appears to have failed:

[#8754](https://circleci.com/gh/cockroachdb/cockroach/8754):

```
I1029 20:15:45.963365 955 raft/raft.go:409  [group 1] 8 became follower at term 1
I1029 20:15:45.966295 955 raft/raft.go:503  [group 1] 8 [term: 1] received a MsgApp message with higher term from 1 [term: 6]
I1029 20:15:45.966408 955 raft/raft.go:409  [group 1] 8 became follower at term 6
W1029 20:15:45.966729 955 multiraft/multiraft.go:1041  failed to look up replica ID for range 1 (disabling replica ID check): storage/store.go:1569: store 3 not found as replica of range 1
I1029 20:15:45.973107 955 storage/store.go:1472  got an event membership change committed event: EndTransaction ["\x00\x00\x00k1\x00\x01rdsc","")
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x10 pc=0xdcd650]

goroutine 6862 [running]:
github.com/coreos/etcd/raft.stepLeader(0xc820462ea0, 0xb, 0x0, 0x8, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/coreos/etcd/raft/raft.go:590 +0x20e0
github.com/coreos/etcd/raft.(*raft).Step(0xc820462ea0, 0xb, 0x0, 0x8, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/coreos/etcd/raft/raft.go:511 +0x380
github.com/coreos/etcd/raft.(*multiNode).run(0xc8205d6a80)
	/go/src/github.com/coreos/etcd/raft/multinode.go:258 +0x1667
created by github.com/coreos/etcd/raft.StartMultiNode
--
	/go/src/github.com/cockroachdb/cockroach/storage/store.go:595 +0x447
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820451aa0, 0xc8203abfc0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:88 +0x60
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x70
FAIL	github.com/cockroachdb/cockroach/storage	19.358s
=== RUN   TestBatchBasics
I1029 20:15:24.220419 949 storage/engine/rocksdb.go:132  closing in-memory rocksdb instance
--- PASS: TestBatchBasics (0.01s)
=== RUN   TestBatchGet
I1029 20:15:24.224933 949 storage/engine/rocksdb.go:132  closing in-memory rocksdb instance
--- PASS: TestBatchGet (0.01s)
=== RUN   TestBatchMerge
I1029 20:15:24.231714 949 storage/engine/rocksdb.go:132  closing in-memory rocksdb instance
--- PASS: TestBatchMerge (0.00s)
=== RUN   TestBatchProto

```
Please assign, take a look and update the issue accordingly.

**Comment by bdarnell** [comment]
May be the same as #3010, although the stack traces are not identical.

**Comment by tamird** [comment]
Is this fixed by #3109?

**Comment by bdarnell** [comment]
No. #3109 only fixed panics with the message "requested entry at index unavailable".
